src/storage/adaptive.rs

```diff
@@ -25,14 +25,17 @@ impl<T> AdaptiveStorageProvider<T>
 where
     T: StorageProvider,
 {
-    /// Creates a new [AdaptiveStorageProvider]. The supplied size is used to construct a
-    /// [BoundedStorageReader] when the stream doesn't have a known content length.
+    /// Creates a new [`AdaptiveStorageProvider`]. The supplied size is used to construct a
+    /// [`BoundedStorageReader`] when the stream doesn't have a known content length.
     pub fn new(inner: T, size: NonZeroUsize) -> Self {
-        Self { inner, size }
+        Self {
```

Comment:
Seems like clippy doesn't like a few of these formatting changes like this one. Once that's fixed I'm good with merging this.

Reply:
had a look at the blame, that was me, I forgot to run the autoformatter after removing a field from the struct (did the refactor in two stages, one added a field which was later removed)